pkg/cloud/ibm/assets/deployment.yaml

```diff
@@ -0,0 +1,114 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: ibm-cloud-controller-manager
+  namespace: openshift-cloud-controller-manager
+  labels:
+    k8s-app: ibm-cloud-controller-manager
+spec:
+  replicas: 3
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 3
+  selector:
+    matchLabels:
+      k8s-app: ibm-cloud-controller-manager
+  template:
+    metadata:
+      labels:
+        k8s-app: ibm-cloud-controller-manager
+    spec:
+      priorityClassName: system-cluster-critical
+      hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - topologyKey: "kubernetes.io/hostname"
+            labelSelector:
+              matchLabels:
+                k8s-app: ibm-cloud-controller-manager
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+        operator: Exists
+      - effect: NoExecute
+        key: node.kubernetes.io/unreachable
+        operator: Exists
+        tolerationSeconds: 120
+      - effect: NoExecute
+        key: node.kubernetes.io/not-ready
+        operator: Exists
+        tolerationSeconds: 120
+      - effect: NoSchedule
+        key: node.cloudprovider.kubernetes.io/uninitialized
+        operator: Exists
+      - effect: NoSchedule
+        key: node.kubernetes.io/not-ready
+        operator: Exists
+      serviceAccountName: cloud-controller-manager
+      terminationGracePeriodSeconds: 90
+      containers:
+      # Container is required to specify name in order for image substitution
+      # to take place. Cloud-controller-manager is substituted with
+      # real CCM image from the cluster
+      - name: cloud-controller-manager
+        image: quay.io/openshift/origin-ibm-cloud-controller-manager
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          runAsGroup: 0
+          runAsNonRoot: true
+        env:
+          - name: VPCCTL_CLOUD_CONFIG
+            value: /etc/kubernetes/cloud.conf
+        command:
+          - /bin/bash
+          - -c
+          - |
+            #!/bin/bash
+            set -o allexport
+            if [[ -f /etc/kubernetes/apiserver-url.env ]]; then
+              source /etc/kubernetes/apiserver-url.env
+            fi
+            exec /bin/ibm-cloud-controller-manager \
+            --port=0 \
```

Comment:
We also need to add `--leader-elect-resource-namespace=openshift-cloud-controller-manager` because by default it uses `kube-system` namespace.

Reply:
updated

Reply:
Why port 0? Default port is 10258, which is reserved for CCMs.
You will need to include a port stanza in the pod spec as well to make the verify task happy

Reply:
@JoelSpeed I believe `--port=0` is disabling the insecure port.  Eventually this option will be removed from Kubernetes for all controllers.  The tentative removal is scheduled for v1.24.

Reply:
Good point, well made! I forgot they have a separate secure port flag, ignore me

> You will need to include a port stanza in the pod spec as well to make the verify task happy

This is still relevant though

Reply:
port stanza added to container